### PR TITLE
Compiler Installation Ubuntu 24 fix

### PR DESCRIPTION
**Pull request by RakeshwarK**
Remove all alternatives for 'cpp' to avoid the symlink looping issue which is causing the error "too many levels of symbolic links" on Ubuntu 24.04.

### CODE REVIEW

**Comment by alexffrank** [comment]
LGTM. Please document the testing done for this bug fix.

**Review by alexffrank — APPROVED** [review]

**Comment by imadityaa** [comment]
The compiler installation is being used by multiple workloads in Linux, can we test the workloads like STREAM, SpecCpu, etc where this is used, with appropriate Linux versions, to make sure that backward compatibility is maintained.

File: src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs [hunk 1]
```diff
@@ -77,6 +77,7 @@ public async Task CompilerInstallationRunsTheExpectedWorkloadCommandInLinuxForGc
                     $"--slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-123 " +
                     $"--slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-123 " +
                     $"--slave /usr/bin/gfortran gfortran /usr/bin/gfortran-123",
+                "sudo update-alternatives --remove-all cpp",
```
[the remark this answers is not in the document]
Comment (RakeshwarK) [reply src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs:80]: We want to remove all symlinks of cpp.

File: src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs [hunk 2]
```diff
@@ -77,6 +77,7 @@ public async Task CompilerInstallationRunsTheExpectedWorkloadCommandInLinuxForGc
...
                 "sudo update-alternatives --install /usr/bin/cpp cpp /usr/bin/cpp-123 1230",
             };
```
[the remark this answers is not in the document]
Comment (RakeshwarK) [reply src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs:83]: Sure Alex, will bring this up.

File: src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs [hunk 3]
```diff
@@ -109,7 +110,7 @@ public async Task CompilerInstallationRunsTheExpectedWorkloadCommandInLinuxForGc
                 await compilerInstallation.ExecuteAsync(CancellationToken.None).ConfigureAwait(false);
             }
 
-            Assert.AreEqual(7, commandExecuted);
+            Assert.AreEqual(8, commandExecuted);
```
[the remark this answers is not in the document]
Comment (RakeshwarK) [reply src/VirtualClient/VirtualClient.Dependencies.UnitTests/CompilerInstallationTests.cs:113]: Sure, will consider this in the next PR.

**Comment by RakeshwarK** [comment]
I have tested this on X64 ubuntu (24,22,20) and Arm Ubuntu (24, 22).

**Review by brdeyo — APPROVED** [review]

**Comment by brdeyo** [comment]
Merging changes to address compiler installation issues on Ubuntu 24.04

**Pull request merged**